### PR TITLE
fixed usage of size_t for time_t (now using difftime()), in order to …

### PR DESCRIPTION
**Pull request by AiyionPrime**
…make it compile on armhf architectures

This allows us to do not care about size_t != time_t errors anymore.
Source:
http://stackoverflow.com/a/22175846/3458484

Tested on:
GNU/Linux 4.4.0-59-generic     x86_64
GNU/Linux 4.4.21-v7+              armv7l

works like a charm.

### CODE REVIEW

File: src/protocol_misc.c [hunk 1]
```diff
@@ -100,7 +100,7 @@ bool send_ping(connection_t *c) {
 	c->status.pinged = true;
 	c->last_ping_time = now;
 
-	return send_request(c, "%d %zu %ld", PING, c->last_ping_time.tv_sec, c->last_ping_time.tv_usec);
+	return send_request(c, "%d %lf %ld", PING, difftime(c->last_ping_time.tv_sec,0), c->last_ping_time.tv_usec);
```
Comment (lemoer) [inline src/protocol_misc.c]: When changing the protocol, please also have a look on ```ping_h()``` and maybe also ```send_ping()``` and ```pong_h()```.

**Review by lemoer — COMMENTED** [review]
Did you take a further look into the protocol?

File: src/protocol_misc.c (hunk 1 above)
Comment (AiyionPrime) [reply src/protocol_misc.c]: Sorry. Thx @lemoer.

Comment (thorkill) [reply src/protocol_misc.c]: @AiyionPrime what architecture you want to support?

**Comment by AiyionPrime** [comment]
review pending for raspberry pi.
works under  GNU/Linux 4.4.0-59-generic x86_64 so far.

**Comment by AiyionPrime** [comment]
> @AiyionPrime what architecture you want to support?

I want to re-support armhf (Ubuntu Phones and Raspberry Pi)

File: src/protocol_misc.c (hunk 1 above)
Comment (AiyionPrime) [reply src/protocol_misc.c]: armhf

**Comment by lemoer** [comment]
Maybe you shoud add to your documentation, that you do not support systems where ```sizeof size_t != sizeof time_t```. :wink:

**Pull request merged**